src/dynamic_boundary_conditions/hyetograph.py

```diff
@@ -1,63 +1,326 @@
+    """
+    Transform the storm length incremental rainfall depths for sites within the catchment area based on
+    the selected hyetograph method and returns in Pandas DataFrame format.
+
+    Parameters
+    ----------
+    storm_length_data : pd.DataFrame
+        Incremental rainfall depths for sites within the catchment area for a specific storm duration.
+    time_to_peak_hrs : int
+        The time in hours when rainfall is at its greatest (reaches maximum).
+    increment_mins : int
+        Time interval in minutes.
+    hyeto_method : Literal["alt_block", "chicago"]
+        Hyetograph method to be used. One of 'alt_block' or 'chicago', i.e., Alternating Block Method or
+        Chicago Method.
+    """
+    hyetograph_sites_data = []
+    for column_num in range(1, len(storm_length_data.columns)):
+        site_data = storm_length_data.iloc[:, [0, column_num]]
+        if hyeto_method == "alt_block":
+            site_data = site_data.sort_values(by=site_data.columns[1], ascending=False)
+        else:
+            site_data_right = site_data.div(2)
+            site_data_left = site_data_right[::-1]
+            site_data = pd.concat([site_data_left, site_data_right]).reset_index(drop=True)
+        site_data = add_time_information(site_data, time_to_peak_hrs, increment_mins, hyeto_method)
+        hyetograph_sites_data.append(site_data)
+    hyetograph_data = pd.concat(hyetograph_sites_data, axis=1, ignore_index=False)
+    hyetograph_data = hyetograph_data.loc[:, ~hyetograph_data.columns.duplicated(keep="last")]
+    return hyetograph_data
+
+
+def get_hyetograph_data(
+        rain_depth_in_catchment: pd.DataFrame,
+        storm_length_hrs: int,
+        time_to_peak_hrs: int,
+        increment_mins: int,
+        interp_method: str,
+        hyeto_method: Literal["alt_block", "chicago"]) -> pd.DataFrame:
+    """
+    Get hyetograph data for all sites within the catchment area and returns in Pandas DataFrame format.
+
+    Parameters
+    ----------
+    rain_depth_in_catchment : pd.DataFrame
+        Rainfall depths for sites within the catchment area for a specified scenario retrieved from the database.
+    storm_length_hrs : int
+        Storm duration in hours.
+    time_to_peak_hrs : int
+        The time in hours when rainfall is at its greatest (reaches maximum).
+    increment_mins : int
+        Time interval in minutes.
+    interp_method : str
+        Temporal interpolation method to be used. One of 'linear', 'nearest', 'nearest-up', 'zero',
+        'slinear', 'quadratic', 'cubic', 'previous', or 'next'.
+    hyeto_method : Literal["alt_block", "chicago"]
+        Hyetograph method to be used. One of 'alt_block' or 'chicago', i.e., Alternating Block Method or
+        Chicago Method.
+    """
+    hyeto_methods = ["alt_block", "chicago"]
+    if hyeto_method not in hyeto_methods:
+        log.error("Invalid hyetograph method.")
+        raise ValueError("Invalid hyetograph method.")
```

Comment:
```suggestion
        raise ValueError(f"Invalid hyetograph method. {hyeto_method} not in {hyeto_methods}")
```

Reply:
good suggestion, amended both log.error and ValueError messages.